tests/app/letter_branding/test_letter_branding_rest.py

```diff
@@ -27,53 +22,110 @@ def test_get_all_letter_brands(client, notify_db_session):
+def test_create_letter_branding_when_user_is_provided(admin_request, sample_user):
+    form = {"name": "super brand", "filename": "super-brand", "created_by_id": str(sample_user.id)}
 
-def test_create_letter_branding(client, notify_db_session):
+    response = admin_request.post(
+        "letter_branding.create_letter_brand",
+        _data=form,
+        _expected_status=201,
+    )
+
+    letter_brand = LetterBranding.query.get(response["id"])
+    assert letter_brand.name == form["name"]
+    assert letter_brand.filename == form["filename"]
+    assert letter_brand.created_at
+    assert letter_brand.updated_at is None
+    assert letter_brand.created_by_id == sample_user.id
+
+
+def test_create_letter_branding_when_user_is_not_provided(admin_request, notify_db_session):
     form = {"name": "super brand", "filename": "super-brand"}
 
-    response = client.post(
-        "/letter-branding",
-        data=json.dumps(form),
-        headers=[("Content-Type", "application/json"), create_admin_authorization_header()],
+    response = admin_request.post(
+        "letter_branding.create_letter_brand",
+        _data=form,
+        _expected_status=201,
+    )
+
+    letter_brand = LetterBranding.query.get(response["id"])
+    assert letter_brand.name == form["name"]
+    assert letter_brand.filename == form["filename"]
+    assert letter_brand.created_at
+    assert letter_brand.updated_at is None
+    assert letter_brand.created_by_id is None
+
+
+def test_update_letter_branding_when_updated_by_user_is_provided(admin_request, sample_user):
+    existing_brand = create_letter_branding()
+
+    form = {"name": "new name", "filename": "new filename", "updated_by_id": str(sample_user.id)}
+
+    response = admin_request.post(
+        "letter_branding.update_letter_branding",
+        letter_branding_id=existing_brand.id,
+        _data=form,
+        _expected_status=201,
+    )
+
+    letter_brand = LetterBranding.query.get(response["id"])
+    assert letter_brand.name == form["name"]
+    assert letter_brand.filename == form["filename"]
+    assert letter_brand.updated_by_id == sample_user.id
+    assert letter_brand.updated_at
+
+
+def test_update_letter_branding_when_updated_by_user_is_not_provided(admin_request, notify_db_session):
+    existing_brand = create_letter_branding()
+
+    form = {"name": "new name", "filename": "new filename"}
+
+    response = admin_request.post(
+        "letter_branding.update_letter_branding",
+        letter_branding_id=existing_brand.id,
+        _data=form,
+        _expected_status=201,
     )
 
-    assert response.status_code == 201
-    json_response = json.loads(response.get_data(as_text=True))
-    letter_brand = LetterBranding.query.get(json_response["id"])
+    letter_brand = LetterBranding.query.get(response["id"])
     assert letter_brand.name == form["name"]
     assert letter_brand.filename == form["filename"]
+    assert letter_brand.updated_by_id is None
+    assert letter_brand.updated_at
```

Comment:
Should we assert the actual timestamp here (with freezegun)?

Reply:
I did consider that when I was writing this, but then decided I was happy with the tests as they are (checking the values have been set without testing the exact values). We mostly don't test these fields at all for other models and testing the exact values would require a change from the `default=datetime.datetime.utcnow` and `onupdate=datetime.datetime.utcnow` way we have of writing things in the models that I prefer.